kratos/utilities/math_utils.h

```diff
@@ -698,9 +698,10 @@ class KRATOS_API(KRATOS_CORE) MathUtils
      * @param rSecondVector Second input vector
      * @return The resulting norm
      */
+    template<class TVector1, class TVector2>
     static inline double Dot(
```

Comment:
```suggestion
    static inline TDataType Dot(
```

Reply:
Do you mean that the temaplte of MathUtils is used for something?

Reply:
idk, but @matekelemen is working on some single precision stuff I think

Reply:
> idk, but @matekelemen is working on some single precision stuff I think

Only in the global (sparse) space, and I haven't seen a dot product there. I'd go with `typename TVector1::value_type` for consistency though.

##

About the entire PR, I'm not sure how useful it is. When it comes to algorithms on arrays of stuff, I think it's much cleaner and more flexible to follow the standard and implement them in terms of iterators.

```cpp
template <class TIterator>
static typename std::iterator_traits<TIterator>::value_type
Dof(TIterator itLeftBegin, TIterator itLeftEnd, TIterator itRightBegin)
```

but then this function is pretty much identical to `std::inner_product`.

Reply:
(@RiccardoRossi is not friend of that)